code/modules/medical/genetics/bioEffects/hemochromia.dm

```diff
@@ -0,0 +1,278 @@
+					if(rand(13) <= 4)
+						switch(H.bioHolder.bloodType)
+							if("A+")
+								if(prob(50))
+									typeRange = 1
+								else
+									typeRange = 2
+							if("A-") typeRange = 3
+							if("B+") typeRange = 4
+							if("B-") typeRange = 5
+							if("AB+") typeRange = 6
+							if("O-") typeRange = 7
+							if("O+") typeRange = 8
+							if("AB-")
+								if(prob(50))
+									typeRange = 9
+								else
+									typeRange = 10
+							else 
+								if(prob(100/3))
+									typeRange = 12
+								else
+									typeRange = 11
+					else
+						typeRange = rand(1,12)
+					var/datum/bioEffect/NEW = null
+					switch(typeRange)
+						if(2) NEW = new /datum/bioEffect/hemochromia/bronze()
+						if(3) NEW = new /datum/bioEffect/hemochromia/gold()
+						if(4) NEW = new /datum/bioEffect/hemochromia/lime()
+						if(5) NEW = new /datum/bioEffect/hemochromia/olive()
+						if(6) NEW = new /datum/bioEffect/hemochromia/jade()
+						if(7) NEW = new /datum/bioEffect/hemochromia/teal()
+						if(8) NEW = new /datum/bioEffect/hemochromia/cobalt()
+						if(9) NEW = new /datum/bioEffect/hemochromia/indigo()
+						if(10) NEW = new /datum/bioEffect/hemochromia/purple()
+						if(11) NEW = new /datum/bioEffect/hemochromia/violet()
+						if(12) NEW = new /datum/bioEffect/hemochromia/fuschia()
+						else NEW = new /datum/bioEffect/hemochromia/rust()
+
+					if(duplicateCheck == 0)
+						H.bioHolder.AddEffectInstance(NEW,1)
+					holder.RemoveEffect(src.id)
+					holder.RemovePoolEffect(src)
+
+/datum/bioEffect/hemochromia
+	name = "Hemochromia Type-A"
+	desc = "An abstract mutation which is so much more unstable than Type-U that it shouldn't even be possible to observe with current technology."
+	id = "hemochromia_abstract"
+	probability = 0
+	effectType = EFFECT_TYPE_MUTANTRACE
+	msgGain = "You feel a bit less soft."
+	msgLose = "You feel lightheaded for a moment."
+	reclaim_fail = 10
+	lockProb = 1
+	lockedGaps = 2
+	lockedDiff = 2
+	lockedChars = list("A","T","C","G")
+	lockedTries = 12
+	stability_loss = 5
+	icon_state  = "hemochromia_unknown"
+	research_level = EFFECT_RESEARCH_DONE
+	occur_in_genepools = 0
+	var/blood_color_R
+	var/blood_color_G
+	var/blood_color_B
+
+	OnLife()
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			if(H.blood_volume < 500 && H.blood_volume > 0)
+				H.blood_volume += 2
+			if(prob(12))
+				H.blood_color = rgb(blood_color_R, blood_color_G, blood_color_B)
+
+	OnRemove()
+		..()
+		if(ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			H.blood_color = DEFAULT_BLOOD_COLOR
+
+/datum/bioEffect/hemochromia/rust
+	name = "Hemochromia Type-R"
+	desc = "Causes the subject's blood cells to take on a scarlet coloration. Also slightly increases blood viscosity."
+	id = "hemochromia_rust"
+	lockProb = 5
+	lockedChars = list("A")
+	icon_state  = "hemochromia_rust"
+
+	OnAdd()
+		..()
+		blood_color_R = rand(254, 46)
+		blood_color_G = rand(0, 0)
+		blood_color_B = rand(0, 2)
```

Comment:
Surprisingly this actually works but the convention is to do `rand(lower_bound, upper_bound)`, makes it more readable. (Also on other heterochromias such lines).

Reply:
Done.